Classes/Private/Manager/Objects/OCTFileInput.m

```diff
@@ -0,0 +1,161 @@
+//
+//  OCTFileInput.m
+//  objcTox
+//
+//  Created by stal on 5/8/2015.
+//  Copyright © 2015 Zodiac Labs. All rights reserved.
+//
+
+#include <sys/stat.h>
+#import "DDLog.h"
+#import "OCTSubmanagerFiles.h"
+
+#undef LOG_LEVEL_DEF
+#define LOG_LEVEL_DEF LOG_LEVEL_DEBUG
+
+@interface OCTFileInput ()
+
+@property (copy)   NSString *path;
+@property (strong) NSFileHandle *readHandle;
+@property NSTimeInterval modifiedTime;
+@property OCTToxFileSize knownFileSize;
+
+@end
+
+@implementation OCTFileInput
+
+- (instancetype)initWithPath:(nonnull NSString *)path
+{
+    self = [super init];
+
+    if (self) {
+        self.path = path;
+        NSError *error = nil;
+        NSDictionary *attrs = [[NSFileManager defaultManager] attributesOfItemAtPath:[path stringByResolvingSymlinksInPath] error:&error];
+
+        if (error) {
```

Comment:
You should check for `attrs` instead of `error`.

> Important Success or failure is indicated by the return value of the method. Although Cocoa methods that indirectly return error objects in the Cocoa error domain are guaranteed to return such objects if the method indicates failure by directly returning nil or NO, you should always check that the return value is nil or NO before attempting to do anything with the NSError object.

https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/CreateCustomizeNSError/CreateCustomizeNSError.html